Add clarification document for the Twins sample (#276)
Azure-Samples/azure-iot-samples-csharp: iot-hub/Samples/device/TwinSample/readme.md

# Device Twins Sample

[![Documentation](../../../../docs/media/docs-link-buttons/azure-documentation.svg)](https://docs.microsoft.com/azure/iot-develop/)

> This sample code demonstrates device twins support interacts with IoT Hub.

Reviewer: ```suggestion
This sample code demonstrates device twins support with IoT hub.
```

Author: Updated

> For more info about Device Twins:  

Reviewer: ```suggestion
For more info about device twins:  
```

> - [Device Twin getting started](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-csharp-csharp-twin-getstarted)

Reviewer: ```suggestion
- [Device twin getting started](https://docs.microsoft.com/azure/iot-hub/iot-hub-csharp-csharp-twin-getstarted)
```

> - [Device Twins Developer Guide](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-device-twins)

Reviewer: ```suggestion
- [Device Twins Developer Guide](https://docs.microsoft.com/azure/iot-hub/iot-hub-devguide-device-twins)
```

> > Note for PnP Device Twin Support with IoT Hub and IoT Central, there specific conventions of the JSON document to follow:

Reviewer: ```suggestion
> Note for Plug and Play device twin support with IoT hub and IoT Central, there specific conventions of the JSON format to follow:
```

> > - [PnP Device developer Guide](https://docs.microsoft.com/en-us/azure/iot-develop/concepts-developer-guide-device?pivots=programming-language-csharp)

Reviewer: ```suggestion
> - [PnP device developer guide](https://docs.microsoft.com/azure/iot-develop/concepts-developer-guide-device?pivots=programming-language-csharp)
```

> > - [PnP Device Samples](https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/main/iot-hub/Samples/device/PnpDeviceSamples)

Reviewer: ```suggestion
> - [PnP device samples](https://github.com/Azure-Samples/azure-iot-samples-csharp/tree/main/iot-hub/Samples/device/PnpDeviceSamples)
```
